src/Avalonia.Controls/NumericUpDown/NumericUpDown.cs

```diff
@@ -408,6 +409,7 @@ protected override void OnApplyTemplate(TemplateAppliedEventArgs e)
             if (TextBox != null)
             {
                 TextBox.Text = Text;
+                TextBox.TabIndex = TabIndex;
```

Comment:
We can revert all the changes and just add one line
```csharp
                TextBox[!TabIndexProperty] = this[!TabIndexProperty];
```

Reply:
Worked with the test so cool with me, thanks!